earthfile2llb/listener.go

```diff
@@ -462,7 +458,7 @@ func (l *listener) ExitSaveArtifact(c *parser.SaveArtifactContext) {
 	saveFrom := l.expandArgs(fs.Args()[0], false)
 	saveTo = l.expandArgs(saveTo, false)
 	saveAsLocalTo = l.expandArgs(saveAsLocalTo, false)
-	err = l.converter.SaveArtifact(l.ctx, saveFrom, saveTo, saveAsLocalTo, *keepTs, *keepOwn, *ifExists)
+	err = l.converter.SaveArtifact(l.ctx, saveFrom, saveTo, saveAsLocalTo, *keepTs, *keepOwn, *ifExists, l.pushOnlyAllowed)
```

Comment:
`pushOnlyAllowed` is here to signify we have encountered a push. Could rename this var; but didn't because im not globally allowing everything like before.

Reply:
If you wanted you could do
```suggestion
	isPushPhase := l.pushOnlyAllowed
	err = l.converter.SaveArtifact(l.ctx, saveFrom, saveTo, saveAsLocalTo, *keepTs, *keepOwn, *ifExists, isPushPhase)
```

Reply:
Ehhhhhhhh that feels like extra ceremony. Not worth it IMO; I would rather rename it to `pushEncountered` across all of converter if I went renaming; since that preserves the business logic meaning well enough

Reply:
🤷 

I like ceremonies :-)